<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>xToolsEmitterToolUiEditor</class>
 <widget class="QWidget" name="xToolsEmitterToolUiEditor">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>482</width>
    <height>400</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string notr="true">Form</string>
  </property>
  <layout class="QGridLayout" name="gridLayout">
   <item row="0" column="0" colspan="3">
    <widget class="QGroupBox" name="groupBox">
     <property name="title">
      <string>Parameters</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_2">
      <item row="2" column="0">
       <widget class="QLabel" name="label_2">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="text">
         <string>Format</string>
        </property>
       </widget>
      </item>
      <item row="4" column="3">
       <widget class="xToolsCrcAlgorithmComboBox" name="comboBoxAlgorithm"/>
      </item>
      <item row="4" column="2">
       <widget class="QLabel" name="label_7">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="text">
         <string>Algorithm</string>
        </property>
       </widget>
      </item>
      <item row="5" column="0">
       <widget class="QLabel" name="label_5">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="text">
         <string>Prefix</string>
        </property>
       </widget>
      </item>
      <item row="4" column="1">
       <widget class="QSpinBox" name="spinBoxInterval">
        <property name="maximum">
         <number>10000</number>
        </property>
       </widget>
      </item>
      <item row="6" column="1">
       <widget class="xToolsAffixesComboBox" name="comboBoxSufix"/>
      </item>
      <item row="3" column="1">
       <widget class="xToolsEscapeCharacterComboBox" name="comboBoxEscaoe"/>
      </item>
      <item row="4" column="0">
       <widget class="QLabel" name="label_4">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="text">
         <string>Interval</string>
        </property>
       </widget>
      </item>
      <item row="5" column="3">
       <widget class="QSpinBox" name="spinBoxEndIndex"/>
      </item>
      <item row="8" column="1" colspan="3">
       <widget class="QLineEdit" name="lineEditData"/>
      </item>
      <item row="9" column="0">
       <widget class="QLabel" name="label">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="text">
         <string>Description</string>
        </property>
       </widget>
      </item>
      <item row="5" column="2">
       <widget class="QLabel" name="label_9">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="text">
         <string>End index</string>
        </property>
       </widget>
      </item>
      <item row="9" column="1" colspan="3">
       <widget class="QLineEdit" name="lineEditDescription">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
       </widget>
      </item>
      <item row="0" column="0" colspan="4">
       <widget class="QCheckBox" name="checkBoxEnable">
        <property name="text">
         <string>Enable</string>
        </property>
       </widget>
      </item>
      <item row="13" column="0" colspan="4">
       <widget class="QLabel" name="label_13">
        <property name="text">
         <string>If the parameters of CRC is set error, all bytes of data will be calculcated.</string>
        </property>
       </widget>
      </item>
      <item row="5" column="1">
       <widget class="xToolsAffixesComboBox" name="comboBoxPrefix"/>
      </item>
      <item row="12" column="0" colspan="4">
       <widget class="QLabel" name="label_12">
        <property name="text">
         <string>The CRC end index is from tail, the last byte of data is 0.</string>
        </property>
       </widget>
      </item>
      <item row="3" column="0">
       <widget class="QLabel" name="label_3">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="text">
         <string>Escape</string>
        </property>
       </widget>
      </item>
      <item row="6" column="3">
       <widget class="QSpinBox" name="spinBoxStartIndex">
        <property name="toolTip">
         <string notr="true">11</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="xToolsTextFormatComboBox" name="comboBoxFormat">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
       </widget>
      </item>
      <item row="8" column="0">
       <widget class="QLabel" name="label_10">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="text">
         <string>Data</string>
        </property>
       </widget>
      </item>
      <item row="3" column="2" colspan="2">
       <widget class="QCheckBox" name="checkBoxBigEndian">
        <property name="text">
         <string>Big endian byte order</string>
        </property>
       </widget>
      </item>
      <item row="6" column="0">
       <widget class="QLabel" name="label_6">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="text">
         <string>Sufix</string>
        </property>
       </widget>
      </item>
      <item row="6" column="2">
       <widget class="QLabel" name="label_8">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="text">
         <string>Start index</string>
        </property>
       </widget>
      </item>
      <item row="2" column="2" colspan="2">
       <widget class="QCheckBox" name="checkBoxCrcEnable">
        <property name="text">
         <string>Append CRC data</string>
        </property>
       </widget>
      </item>
      <item row="11" column="0" colspan="4">
       <widget class="QLabel" name="label_11">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="text">
         <string>The CRC start index is start form header, the first byte is 0.</string>
        </property>
        <property name="wordWrap">
         <bool>false</bool>
        </property>
       </widget>
      </item>
      <item row="10" column="0">
       <widget class="QLabel" name="label_14">
        <property name="styleSheet">
         <string notr="true">QLabel{color:red}</string>
        </property>
        <property name="text">
         <string>Note:</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="1" column="0">
    <spacer name="horizontalSpacer">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>40</width>
       <height>20</height>
      </size>
     </property>
    </spacer>
   </item>
   <item row="1" column="1">
    <widget class="QPushButton" name="pushButtonCancel">
     <property name="text">
      <string>Cancel</string>
     </property>
    </widget>
   </item>
   <item row="1" column="2">
    <widget class="QPushButton" name="pushButtonApply">
     <property name="text">
      <string>Apply</string>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>xToolsEscapeCharacterComboBox</class>
   <extends>QComboBox</extends>
   <header location="global">xToolsEscapeCharacterComboBox.h</header>
  </customwidget>
  <customwidget>
   <class>xToolsTextFormatComboBox</class>
   <extends>QComboBox</extends>
   <header location="global">xToolsTextFormatComboBox.h</header>
  </customwidget>
  <customwidget>
   <class>xToolsAffixesComboBox</class>
   <extends>QComboBox</extends>
   <header location="global">xToolsAffixesComboBox.h</header>
  </customwidget>
  <customwidget>
   <class>xToolsCrcAlgorithmComboBox</class>
   <extends>QComboBox</extends>
   <header location="global">xToolsCrcAlgorithmComboBox.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
